src/d/actor/d_a_toge.cpp

```diff
@@ -4,104 +4,334 @@
+
+    mpBgW1->Move();
+    mpBgW2->Move();
+
+    return TRUE;
 }
 
 /* 000004F4-00000620       .text _create__8daToge_cFv */
-s32 daToge_c::_create() {
-    /* Nonmatching */
+BOOL daToge_c::_create() {
+    fopAcM_SetupActor(this, daToge_c);
+
+    s32 ret = dComIfG_resLoad(&m_Phs, m_arcname);
+
+    if (ret == cPhs_COMPLEATE_e) {
+        if (!fopAcM_entrySolidHeap(this, CheckCreateHeap, 0x1400)) {
+            return cPhs_ERROR_e;
+        }
+
+        Create();
+    }
+
+    return ret;
 }
 
 /* 000007D8-00000868       .text set_mtx__8daToge_cFv */
 void daToge_c::set_mtx() {
-    /* Nonmatching */
+    mpModel->setBaseScale(scale);
+    mDoMtx_stack_c::transS(current.pos.x, current.pos.y, current.pos.z);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
+    mDoMtx_copy(mDoMtx_stack_c::get(), mtx1);
 }
 
 /* 00000868-000009F4       .text _execute__8daToge_cFv */
 BOOL daToge_c::_execute() {
-    /* Nonmatching */
+    if (mSwitchNo != 0xFF) {
+        if (fopAcM_isSwitch(this, mSwitchNo) != 0) {
+            if (mEventState != 1) {
+                this->unk485 = 1;
+            }
+        } else if (fopAcM_isSwitch(this, mSwitchNo) == 0 && mEventState != 4) {
+            mEventState = 3;
+        }
+    } else if (mSwitchNo == 0xFF) {  // Redundant condition (necessary for matching)
+        search_wind();
+
+        daWindTag::daWindTag_c* pActor = (daWindTag::daWindTag_c*)fopAcM_SearchByID(mWindTagId);
+
+        if (pActor != NULL) {
+            if (pActor->unk498 > 0.0f) {
+                if (mEventState != 4) {
+                    mEventState = 3;
+                }
+            } else if (mEventState != 1) {
+                this->unk485 = 1;
+            }
+        }
+    }
+
+    toge_move();
+
+    mpModel->setBaseScale(scale);
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
+    mDoMtx_copy(mDoMtx_stack_c::get(), mtx1);
+
+    mpBgW2->Move();
+
+    set_collision();
+
+    return TRUE;
 }
 
 /* 000009F4-00000A78       .text set_collision__8daToge_cFv */
 void daToge_c::set_collision() {
-    /* Nonmatching */
+    if (mEventState != 2) {
+        cXyz center;
+        center.x = current.pos.x;
+        center.y = current.pos.y;
+        center.z = current.pos.z;
+
+        center.y += (this->unk470 - 10.0f);
+
+        mCyl.SetC(center);
+        dComIfG_Ccsp()->Set(&mCyl);
+    }
 }
 
 /* 00000A78-00000AE0       .text search_wind__8daToge_cFv */
 void daToge_c::search_wind() {
-    /* Nonmatching */
+    fopAc_ac_c* pActor = fopAcM_SearchByName(PROC_WindTag);
+
+    if (pActor != NULL) {
+        mWindTagId = fopAcM_GetID(pActor);
+    } else {
+        mWindTagId = fpcM_ERROR_PROCESS_ID_e;
+    }
 }
 
 /* 00000AE0-00000C1C       .text toge_move__8daToge_cFv */
 void daToge_c::toge_move() {
-    /* Nonmatching */
+    u8 r30 = 1;
+    switch (mEventState)  // Irregular switch
```

Comment:
what's irregular about it?

Reply:
That comment was from the decompiler because `case 2` was placed within the `case 1` block, but I do find it strange for it to have so many fallthroughs and explicitly handle 0.

I'll remove the comment since it's outdated